packages/notification-channel/src/api/index.ts

```diff
@@ -54,4 +56,31 @@ export class NotificationChannelClient extends BaseApiClient {
       return await api.put(`${PATH}/${type}/${id}`, notification);
     });
   }
+
+  public verifySignature(params: {
+    signature: string;
+    secret: string;
+    timestamp: string;
+    validityInMs?: number;
+  }): boolean {
+    if (!params.secret) throw new Error('Secret is missing');
+    if (!params.signature) throw new Error('Signature is missing');
+    if (!params.timestamp) throw new Error('Timestamp is missing');
+
+    try {
+      const TEN_MINUTES_IN_MS = 1000 * 60 * 10;
+      const validityInMillis = params.validityInMs || TEN_MINUTES_IN_MS;
+      const createdAt = new Date(params.timestamp);
+
+      const millisSince = getMillisSince(createdAt);
+      const isRecent = millisSince <= validityInMillis && millisSince >= 0;
```

Comment:
Supernit: I was thinking of reversing the boolean and naming it `isExpired` instead, I find it more intuitive. But consider this a nit and feel free to ignore.